mypy/nodes.py

```diff
@@ -2122,6 +2127,7 @@ def calculate_mro(self) -> None:
         self.is_enum = self._calculate_is_enum()
         # The property of falling back to Any is inherited.
         self.fallback_to_any = any(baseinfo.fallback_to_any for baseinfo in self.mro)
+        self.reset_subtype_cache()
```

Comment:
I don't fully understand why this is necessary (though it certainly isn't /wrong/). Adding a new subclass shouldn't invalidate any of the caches, since they store only positive information?

Reply:
I think that some changes in generic base classes may require this. Say, originally the base class was `C[A]` and after update it is `C[B]`. This would change the subtype relation even though the MRO is unchanged. If we are doing a refresh of the class there would be no AST merge so without this the subtype cache would not get emptied.